wavelength-plugins/badges.js

```diff
@@ -1,134 +1,148 @@
-			this.logModCommand(user.name + " gave the badge '" + selectedBadge + "' badge to " + userid + ".");
-			this.sendReply("The '" + selectedBadge + "' badge was given to '" + userid + "'.");
-			break;
-		case 'create':
+			if (Users.get(targetUser)) Users.get(userid).popup(`|modal||html|You have received a badge from ${WL.nameColor(user.name, true)}: <img src="${Db.badgeData.get(selectedBadge)[1]}" width="16" height="16"> (${selectedBadge})`);
+			this.modlog(`BADGES`, targetUser, `gave badge`);
+			this.sendReply(`The "${selectedBadge}" badge was given to ${parts[0]}.`);
+		},
+
+		create: function (target, room, user) {
 			if (!this.can('ban')) return false;
-			if (parts.length !== 4) return this.errorReply("Correct command: `/badges create, [badge name], [description], [image]`.");
-			let badgeName = Chat.escapeHTML(parts[1].trim());
-			let description = Chat.escapeHTML(parts[2].trim());
-			let img = parts[3].trim();
+			let parts = target.split(',');
+			if (parts.length !== 3) return this.parse("/badgeshelp");
+			let badgeName = Chat.escapeHTML(parts[0].trim());
+			let description = Chat.escapeHTML(parts[1].trim());
+			let img = parts[2].trim();
 			if (Db.badgeData.has(badgeName)) return this.errorReply('This badge already exists.');
 			Db.badgeData.set(badgeName, [description, img]);
-			this.logModCommand(user.name + " created the badge '" + badgeName + ".");
-			Users.get(user.userid).popup('|modal||html|You have successfully created the badge ' + '<img src ="' + img + '" width="16" height="16"> (' + badgeName + ')');
-			break;
-		case 'list':
+			this.modlog(`BADGE`, null, `added badge.`);
+			Users.get(user.userid).popup(`|modal||html|You have successfully created the badge <img src ="${img}" width="16" height="16"> (${badgeName})`);
+		},
+
+		list: function (target, room, user) {
 			if (!this.runBroadcast()) return;
-			output = '<table>';
+			let output = '';
+			output = `<table>`;
 			Db.badgeData.keys().forEach(badge => {
 				let badgeData = Db.badgeData.get(badge);
-				output += '<tr ' + TR_CSS + '> <td ' + TD_CSS + '>' + badgeImg(badgeData[1], badge) + '</td> <td ' + TD_CSS + '>' + badge + '</td> <td ' + TD_CSS + '>' + badgeData[0] + '</td></tr>';
+				output += `<tr ${TR_CSS}> <td ${TD_CSS}>${badgeImg(badgeData[1], badge)}</td> <td ${TD_CSS}>${badge}</td> <td ${TD_CSS}>${badgeData[0]}</td></tr>`;
 			});
-			output += '</table>';
-			this.sendReply('|html|<div class = "infobox' + (this.broadcasting ? '-limited' : '') + '">' + output + '</div>');
-			break;
-		case 'info':
+			output += `</table>`;
+			this.sendReply(`|html|<div class = "infobox ${(this.broadcasting ? '-limited' : '')}">${output}</div>`);
+		},
+
+		info: function (target, room, user) {
 			if (!this.runBroadcast()) return;
+			let selectedBadge;
+			let parts = target.split(',');
```

Comment:
Why does a user have to use `/badges info ,[badge]`? You should just be checking `target` itself.